Modules/Loadable/InteractiveTubesToTree/Logic/vtkSlicerInteractiveTubesToTreeLogic.cxx

```diff
@@ -487,54 +485,117 @@ ::deleteTubeFromSpatialObject(vtkMRMLSpatialObjectsNode* spatialNode, std::set<i
-
+      //currTube is the "to be deleted" tube
+      //update parent id of the children of "to be deleted" tube in the list of tubes under spatial object.
       for (TubeNetType::ChildrenListType::iterator tubeIt_1 = tubeList->begin(); tubeIt_1 != tubeList->end(); ++tubeIt_1)
       {
-        VesselTubeType* currTube_1 =
+        VesselTubeType* currTube1 =
           dynamic_cast<VesselTubeType*>((*tubeIt_1).GetPointer());
-        int tubeId = currTube_1->GetId();
-        it = childrenIdList.find(tubeId);
-        if(it !=  childrenIdList.end())
+        if(currTube1 && currTube1->GetParentId() == currTubeId)
         {
-          currTube_1->SetParentId(parentID);
+          currTube1->SetParentId(spatialObject->GetId());
         }
       }
-
       //delete "to be deleted" tube from children of its parent.
-      TubeNetType::ChildrenListType* currTubeParentChildrenList = currTube->GetParent()->GetChildren(1, childName);
+      TubeNetType::ChildrenListType* currTubeParentChildrenList = currTube->GetParent()->GetChildren(0, childName);
       for (TubeNetType::ChildrenListType::iterator tubeIt2 = currTubeParentChildrenList->begin();
         tubeIt2 != currTubeParentChildrenList->end(); ++tubeIt2)
       {
-        VesselTubeType* currTube2 =
+        VesselTubeType* currTube1 =
       dynamic_cast<VesselTubeType*>((*tubeIt2).GetPointer());
-        if (!currTube || currTube->GetNumberOfPoints() < 2)
+        if (!currTube1 || currTube1->GetNumberOfPoints() < 1)
         {
           continue;
         }
-        if(currTube2->GetId() == currTubeId)
+        if(currTube1->GetId() == currTubeId)
         {
-          currTube2->Clear();
+          currTube1->Clear();
           break;
         }        
-      }
-    }  
+      } 
+ //     spatialObject->RemoveSpatialObject(currTube);
+    }
   }
   spatialNode->UpdatePolyDataFromSpatialObject();
+}
+
+//---------------------------------------------------------------------------=
+bool vtkSlicerInteractiveTubesToTreeLogic
+::GetSpatialObjectOrphanStatusData(vtkMRMLSpatialObjectsNode* spatialNode, int currTubeID)
+{
+  if (!spatialNode)
+  {
+    return -1;
+  }
+  TubeNetType* spatialObject = spatialNode->GetSpatialObject();
+
+  char childName[] = "Tube";
+  TubeNetType::ChildrenListType* tubeList =
+    spatialObject->GetChildren(spatialObject->GetMaximumDepth(), childName);
+
+  for (TubeNetType::ChildrenListType::iterator tubeIt = tubeList->begin(); tubeIt != tubeList->end(); ++tubeIt)
+  {
+    VesselTubeType* currTube =
+      dynamic_cast<VesselTubeType*>((*tubeIt).GetPointer());
+    if (!currTube || currTube->GetNumberOfPoints() < 1)
+    {
+      continue;
+    }
+    if (currTube->GetId() == currTubeID)
+    {
+      if(currTube->GetChildren(0, childName)->size() > 0)
+      {
+        return false;
+      }
+      if(currTube->HasParent() && currTube->GetParentId() != spatialObject->GetId())
+      {
+        return false;
+      }
+      break;
+    }
+  }
+  return true;
```

Comment:
This returns true even if `currTubeID` is not a child of `spatialNode`. Is that ok?